### PR TITLE
python311Packages.autobahn: 22.7.1 -> 23.6.2

### PR DESCRIPTION
**Pull request by hellwolf**
###### Description of changes

While updating python311Packages.autobahn version, also fixing a future blocker identified in https://github.com/NixOS/nixpkgs/pull/240007#issuecomment-1622038374

* Bump version to 23.6.2
* Fixes:
 - remove the problematic extra dependency of `py-eth-sig-utils`
 - `disabled = pythonOlder "3.9"` in accordance with the upstream

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hellwolf** [comment]
51 packages built:
baserow baserow.dist buildbot buildbot-full buildbot-ui buildbot-worker buildbot-worker.dist buildbot.dist gnome-keysign gnome-keysign.dist magic-wormhole magic-wormhole.dist paperless-ngx python310Packages.autobahn python310Packages.autobahn.dist python310Packages.channels python310Packages.channels-redis python310Packages.channels-redis.dist python310Packages.channels.dist python310Packages.daphne python310Packages.daphne.dist python310Packages.labgrid python310Packages.labgrid.dist python310Packages.magic-wormhole-mailbox-server python310Packages.magic-wormhole-mailbox-server.dist python310Packages.magic-wormhole-transit-relay python310Packages.magic-wormhole-transit-relay.dist python310Packages.strawberry-graphql python310Packages.strawberry-graphql.dist python311Packages.autobahn python311Packages.autobahn.dist python311Packages.channels python311Packages.channels-redis python311Packages.channels-redis.dist python311Packages.channels.dist python311Packages.daphne python311Packages.daphne.dist python311Packages.labgrid python311Packages.labgrid.dist python311Packages.magic-wormhole python311Packages.magic-wormhole-mailbox-server python311Packages.magic-wormhole-mailbox-server.dist python311Packages.magic-wormhole-transit-relay python311Packages.magic-wormhole-transit-relay.dist python311Packages.magic-wormhole.dist python311Packages.strawberry-graphql python311Packages.strawberry-graphql.dist tahoe-lafs tahoe-lafs.dist tahoe-lafs.doc tahoe-lafs.info

Result of `nixpkgs-review pr 241713` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>51 packages built:</summary>
  <ul>
    <li>baserow</li>
    <li>baserow.dist</li>
    <li>buildbot</li>
    <li>buildbot-full</li>
    <li>buildbot-ui</li>
    <li>buildbot-worker</li>
    <li>buildbot-worker.dist</li>
    <li>buildbot.dist</li>
    <li>gnome-keysign</li>
    <li>gnome-keysign.dist</li>
    <li>magic-wormhole (python310Packages.magic-wormhole)</li>
    <li>magic-wormhole.dist (python310Packages.magic-wormhole.dist)</li>
    <li>paperless-ngx</li>
    <li>python310Packages.autobahn</li>
    <li>python310Packages.autobahn.dist</li>
    <li>python310Packages.channels</li>
    <li>python310Packages.channels-redis</li>
    <li>python310Packages.channels-redis.dist</li>
    <li>python310Packages.channels.dist</li>
    <li>python310Packages.daphne</li>
    <li>python310Packages.daphne.dist</li>
    <li>python310Packages.labgrid</li>
    <li>python310Packages.labgrid.dist</li>
    <li>python310Packages.magic-wormhole-mailbox-server</li>
    <li>python310Packages.magic-wormhole-mailbox-server.dist</li>
    <li>python310Packages.magic-wormhole-transit-relay</li>
    <li>python310Packages.magic-wormhole-transit-relay.dist</li>
    <li>python310Packages.strawberry-graphql</li>
    <li>python310Packages.strawberry-graphql.dist</li>
    <li>python311Packages.autobahn</li>
    <li>python311Packages.autobahn.dist</li>
    <li>python311Packages.channels</li>
    <li>python311Packages.channels-redis</li>
    <li>python311Packages.channels-redis.dist</li>
    <li>python311Packages.channels.dist</li>
    <li>python311Packages.daphne</li>
    <li>python311Packages.daphne.dist</li>
    <li>python311Packages.labgrid</li>
    <li>python311Packages.labgrid.dist</li>
    <li>python311Packages.magic-wormhole</li>
    <li>python311Packages.magic-wormhole-mailbox-server</li>
    <li>python311Packages.magic-wormhole-mailbox-server.dist</li>
    <li>python311Packages.magic-wormhole-transit-relay</li>
    <li>python311Packages.magic-wormhole-transit-relay.dist</li>
    <li>python311Packages.magic-wormhole.dist</li>
    <li>python311Packages.strawberry-graphql</li>
    <li>python311Packages.strawberry-graphql.dist</li>
    <li>tahoe-lafs</li>
    <li>tahoe-lafs.dist</li>
    <li>tahoe-lafs.doc</li>
    <li>tahoe-lafs.info</li>
  </ul>
</details>

File: pkgs/development/python-modules/autobahn/default.nix [hunk 1]
```diff
@@ -49,22 +49,14 @@
...
 buildPythonPackage rec {
   pname = "autobahn";
-  version = "22.7.1";
+  version = "23.6.2";
   format = "setuptools";
 
-  disabled = pythonOlder "3.7";
-
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/autobahn/default.nix:58]: Please remove the unused fetchpatch from inputs.

File: pkgs/development/python-modules/autobahn/default.nix [hunk 2]
```diff
@@ -110,7 +102,7 @@ buildPythonPackage rec {
     serialization = [ cbor2 flatbuffers msgpack ujson py-ubjson ];
     twisted = [ attrs args.twisted zope_interface ];
     ui = [ pygobject3 ];
-    xbr = [ base58 cbor2 click ecdsa eth-abi jinja2 hkdf mnemonic py-ecc py-eth-sig-utils py-multihash rlp spake2 twisted /* web3 xbr */ yapf /* zlmdb */ ];
+    xbr = [ base58 cbor2 click ecdsa eth-abi jinja2 hkdf mnemonic py-ecc py-multihash rlp spake2 twisted /* web3 py-eth-sig-utils xbr */ yapf /* zlmdb */ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/autobahn/default.nix:105]:
```suggestion
    xbr = [ base58 cbor2 click ecdsa eth-abi jinja2 hkdf mnemonic py-ecc /* py-eth-sig-utils */ py-multihash rlp spake2 twisted /* web3 xbr */ yapf /* zlmdb */ ];
```
those are supposed to be sorted

File: pkgs/development/python-modules/autobahn/default.nix [hunk 3]
```diff
@@ -40,6 +39,7 @@
 , txaio
 , ujson
   # , web3
+  # , py-eth-sig-utils
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/autobahn/default.nix:42]:
```suggestion
```

File: pkgs/development/python-modules/autobahn/default.nix [hunk 4]
```diff
@@ -21,7 +21,6 @@
 , msgpack
 , passlib
 , py-ecc
-, py-eth-sig-utils
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/autobahn/default.nix:24]:
```suggestion
  # , py-eth-sig-utils
```

Comment (hellwolf) [reply pkgs/development/python-modules/autobahn/default.nix:24]: done

File: pkgs/development/python-modules/autobahn/default.nix (hunk 1 above)
Comment (hellwolf) [reply pkgs/development/python-modules/autobahn/default.nix:58]: done

File: pkgs/development/python-modules/autobahn/default.nix (hunk 2 above)
Comment (hellwolf) [reply pkgs/development/python-modules/autobahn/default.nix]: okay

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 241713` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
51 packages built:
</summary>
<ul>
<li>baserow</li>
<li>baserow.dist</li>
<li>buildbot</li>
<li>buildbot-full</li>
<li>buildbot-ui</li>
<li>buildbot-worker</li>
<li>buildbot-worker.dist</li>
<li>buildbot.dist</li>
<li>gnome-keysign</li>
<li>gnome-keysign.dist</li>
<li>magic-wormhole (python310Packages.magic-wormhole)</li>
<li>magic-wormhole.dist (python310Packages.magic-wormhole.dist)</li>
<li>paperless-ngx</li>
<li>python310Packages.autobahn</li>
<li>python310Packages.autobahn.dist</li>
<li>python310Packages.channels</li>
<li>python310Packages.channels-redis</li>
<li>python310Packages.channels-redis.dist</li>
<li>python310Packages.channels.dist</li>
<li>python310Packages.daphne</li>
<li>python310Packages.daphne.dist</li>
<li>python310Packages.labgrid</li>
<li>python310Packages.labgrid.dist</li>
<li>python310Packages.magic-wormhole-mailbox-server</li>
<li>python310Packages.magic-wormhole-mailbox-server.dist</li>
<li>python310Packages.magic-wormhole-transit-relay</li>
<li>python310Packages.magic-wormhole-transit-relay.dist</li>
<li>python310Packages.strawberry-graphql</li>
<li>python310Packages.strawberry-graphql.dist</li>
<li>python311Packages.autobahn</li>
<li>python311Packages.autobahn.dist</li>
<li>python311Packages.channels</li>
<li>python311Packages.channels-redis</li>
<li>python311Packages.channels-redis.dist</li>
<li>python311Packages.channels.dist</li>
<li>python311Packages.daphne</li>
<li>python311Packages.daphne.dist</li>
<li>python311Packages.labgrid</li>
<li>python311Packages.labgrid.dist</li>
<li>python311Packages.magic-wormhole</li>
<li>python311Packages.magic-wormhole-mailbox-server</li>
<li>python311Packages.magic-wormhole-mailbox-server.dist</li>
<li>python311Packages.magic-wormhole-transit-relay</li>
<li>python311Packages.magic-wormhole-transit-relay.dist</li>
<li>python311Packages.magic-wormhole.dist</li>
<li>python311Packages.strawberry-graphql</li>
<li>python311Packages.strawberry-graphql.dist</li>
<li>tahoe-lafs</li>
<li>tahoe-lafs.dist</li>
<li>tahoe-lafs.doc</li>
<li>tahoe-lafs.info</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
buildbot.dist:
<details>

got build log for '/nix/store/216v29cfashd7vki9nl3jsrqdmxssy7n-buildbot-3.8.0-dist' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/216v29cfashd7vki9nl3jsrqdmxssy7n-buildbot-3.8.0-dist' from 'daemon'
When evaluating attribute ‘buildbot.dist’:
warning: unused-argument
Unused argument: `buildPythonPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:3:3:

```
  |
3 | , buildPythonPackage
  |   ^
```
warning: unused-argument
Unused argument: `callPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:36:3:

```
   |
36 | , callPackage
   |   ^
```
</details>
buildbot-full:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/ar798a95hc1nf96lvk06p01a0nyxba92-buildbot-with-plugins-3.8.0' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/ar798a95hc1nf96lvk06p01a0nyxba92-buildbot-with-plugins-3.8.0' from 'daemon'
When evaluating attribute ‘buildbot-full’:
warning: unused-argument
Unused argument: `buildPythonPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:3:3:

```
  |
3 | , buildPythonPackage
  |   ^
```
warning: unused-argument
Unused argument: `callPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:36:3:

```
   |
36 | , callPackage
   |   ^
```
</details>
buildbot:
<details>

got build log for '/nix/store/pq5ls4nw741fpy6kw8z2nkkfjwa9xc1h-buildbot-3.8.0' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/pq5ls4nw741fpy6kw8z2nkkfjwa9xc1h-buildbot-3.8.0' from 'daemon'
When evaluating attribute ‘buildbot’:
warning: unused-argument
Unused argument: `buildPythonPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:3:3:

```
  |
3 | , buildPythonPackage
  |   ^
```
warning: unused-argument
Unused argument: `callPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:36:3:

```
   |
36 | , callPackage
   |   ^
```
</details>
buildbot-ui:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

got build log for '/nix/store/fknikxqkpbwd2lkn9wyy1d3bmww74vnq-buildbot-with-plugins-3.8.0' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/fknikxqkpbwd2lkn9wyy1d3bmww74vnq-buildbot-with-plugins-3.8.0' from 'daemon'
When evaluating attribute ‘buildbot-ui’:
warning: unused-argument
Unused argument: `buildPythonPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:3:3:

```
  |
3 | , buildPythonPackage
  |   ^
```
warning: unused-argument
Unused argument: `callPackage`.
Near pkgs/development/tools/continuous-integration/buildbot/master.nix:36:3:

```
   |
36 | , callPackage
   |   ^
```
</details>
python310Packages.autobahn.dist:
<details>

got build log for '/nix/store/6xclsmdj2nj0j909jbxd4ib0mqv4ph5l-python3.10-autobahn-23.6.2-dist' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/6xclsmdj2nj0j909jbxd4ib0mqv4ph5l-python3.10-autobahn-23.6.2-dist' from 'daemon'
When evaluating attribute ‘python310Packages.autobahn.dist’:
warning: unused-argument
Unused argument: `eth-account`.
Near pkgs/development/python-modules/autobahn/default.nix:13:3:

```
   |
13 | , eth-account
   |   ^
```
</details>
python310Packages.autobahn:
<details>

got build log for '/nix/store/a1a3hri6mrxl1ljmm8dmmw6mxyl8f8im-python3.10-autobahn-23.6.2' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/a1a3hri6mrxl1ljmm8dmmw6mxyl8f8im-python3.10-autobahn-23.6.2' from 'daemon'
When evaluating attribute ‘python310Packages.autobahn’:
warning: unused-argument
Unused argument: `eth-account`.
Near pkgs/development/python-modules/autobahn/default.nix:13:3:

```
   |
13 | , eth-account
   |   ^
```
</details>
python310Packages.daphne.dist:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

</details>
python310Packages.daphne:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

</details>
python310Packages.magic-wormhole-mailbox-server.dist:
<details>

got build log for '/nix/store/gl5wkqzs947j66xl323bdshwvsflfqzg-python3.10-magic-wormhole-mailbox-server-0.4.1-dist' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/gl5wkqzs947j66xl323bdshwvsflfqzg-python3.10-magic-wormhole-mailbox-server-0.4.1-dist' from 'daemon'
When evaluating attribute ‘python310Packages.magic-wormhole-mailbox-server.dist’:
warning: unused-argument
Unused argument: `pyopenssl`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:8:3:

```
  |
8 | , pyopenssl
  |   ^
```
warning: unused-argument
Unused argument: `service-identity`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:9:3:

```
  |
9 | , service-identity
  |   ^
```
</details>
python310Packages.magic-wormhole-mailbox-server:
<details>

got build log for '/nix/store/63acaiska5p3xm0ww5gf52yyvq4b1a2m-python3.10-magic-wormhole-mailbox-server-0.4.1' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/63acaiska5p3xm0ww5gf52yyvq4b1a2m-python3.10-magic-wormhole-mailbox-server-0.4.1' from 'daemon'
When evaluating attribute ‘python310Packages.magic-wormhole-mailbox-server’:
warning: unused-argument
Unused argument: `pyopenssl`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:8:3:

```
  |
8 | , pyopenssl
  |   ^
```
warning: unused-argument
Unused argument: `service-identity`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:9:3:

```
  |
9 | , service-identity
  |   ^
```
</details>
python311Packages.autobahn.dist:
<details>

got build log for '/nix/store/1n1ayf8sq91mh0lbnrm6q0c92apnx1fm-python3.11-autobahn-23.6.2-dist' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/1n1ayf8sq91mh0lbnrm6q0c92apnx1fm-python3.11-autobahn-23.6.2-dist' from 'daemon'
When evaluating attribute ‘python311Packages.autobahn.dist’:
warning: unused-argument
Unused argument: `eth-account`.
Near pkgs/development/python-modules/autobahn/default.nix:13:3:

```
   |
13 | , eth-account
   |   ^
```
</details>
python311Packages.autobahn:
<details>

got build log for '/nix/store/ajldgjg9z5izwm700j7fys0mcyi0w8fw-python3.11-autobahn-23.6.2' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/ajldgjg9z5izwm700j7fys0mcyi0w8fw-python3.11-autobahn-23.6.2' from 'daemon'
When evaluating attribute ‘python311Packages.autobahn’:
warning: unused-argument
Unused argument: `eth-account`.
Near pkgs/development/python-modules/autobahn/default.nix:13:3:

```
   |
13 | , eth-account
   |   ^
```
</details>
python311Packages.daphne.dist:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

</details>
python311Packages.daphne:
<details>

Package is missing maintainers.
If the package is using runCommand please make sure to inherit or list one or more maintainers.

</details>
python311Packages.magic-wormhole-mailbox-server.dist:
<details>

got build log for '/nix/store/dyp8d4y8spgawhi7vga4h5hwhpvn99i6-python3.11-magic-wormhole-mailbox-server-0.4.1-dist' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/dyp8d4y8spgawhi7vga4h5hwhpvn99i6-python3.11-magic-wormhole-mailbox-server-0.4.1-dist' from 'daemon'
When evaluating attribute ‘python311Packages.magic-wormhole-mailbox-server.dist’:
warning: unused-argument
Unused argument: `pyopenssl`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:8:3:

```
  |
8 | , pyopenssl
  |   ^
```
warning: unused-argument
Unused argument: `service-identity`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:9:3:

```
  |
9 | , service-identity
  |   ^
```
</details>
python311Packages.magic-wormhole-mailbox-server:
<details>

got build log for '/nix/store/wilmllb2fqpn2zclv34l6qxq286ylgjm-python3.11-magic-wormhole-mailbox-server-0.4.1' from 'daemon'
warning: 'https://hydra.supersandro.de' does not appear to be a binary cache
got build log for '/nix/store/wilmllb2fqpn2zclv34l6qxq286ylgjm-python3.11-magic-wormhole-mailbox-server-0.4.1' from 'daemon'
When evaluating attribute ‘python311Packages.magic-wormhole-mailbox-server’:
warning: unused-argument
Unused argument: `pyopenssl`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:8:3:

```
  |
8 | , pyopenssl
  |   ^
```
warning: unused-argument
Unused argument: `service-identity`.
Near pkgs/development/python-modules/magic-wormhole-mailbox-server/default.nix:9:3:

```
  |
9 | , service-identity
  |   ^
```
</details>
</details>

**Pull request merged**